pkg/trait/trait_types.go

```diff
@@ -511,8 +510,9 @@ func (e *Environment) configureVolumesAndMounts(vols *[]corev1.Volume, mnts *[]c
 	// Resources (likely application properties or kamelets)
 	if e.Resources != nil {
 		e.Resources.VisitConfigMap(func(configMap *corev1.ConfigMap) {
-			// Camel properties
-			if configMap.Labels[kubernetes.ConfigMapTypeLabel] == "camel-properties" {
+			switch configMap.Labels[kubernetes.ConfigMapTypeLabel] {
+			case "camel-properties":
```

Comment:
ideally this should be a const/enum

Reply:
It is not part of this PR. I've just changed from an if to a switch here. We can make a follow up work to do the required cleaning.

Reply:
Remove as well.